### PR TITLE
JDK-8322572: AllocationMergesTests.java fails with "IRViolationException: There were one or multiple IR rule failures."

### PR DESCRIPTION
**Pull request by JohnTortugo**
Please review this PR to fix a test in `AllocationMergesTest.java`. The test was failing intermittently because of a random value used as parameter was causing some randomization in the shape method's IR graph and consequently the optimization tested wasn't happening.

Tested this locally on Mac, Win and Linux x86_64.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8322572](https://bugs.openjdk.org/browse/JDK-8322572): AllocationMergesTests.java fails with "IRViolationException: There were one or multiple IR rule failures." (**Bug** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17469/head:pull/17469` \
`$ git checkout pull/17469`

Update a local copy of the PR: \
`$ git checkout pull/17469` \
`$ git pull https://git.openjdk.org/jdk.git pull/17469/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17469`

View PR using the GUI difftool: \
`$ git pr show -t 17469`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17469.diff">https://git.openjdk.org/jdk/pull/17469.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17469#issuecomment-1897154036)

### CODE REVIEW

File: test/hotspot/jtreg/compiler/c2/irTests/scalarReplacement/AllocationMergesTests.java [hunk 1]
```diff
@@ -117,11 +117,11 @@ public void runner(RunInfo info) {
...
         Asserts.assertEQ(testLoadAfterTrap_Interp(cond1, x, y),                     testLoadAfterTrap_C2(cond1, x, y));
         Asserts.assertEQ(testCondAfterMergeWithNull_Interp(cond1, cond2, x, y),     testCondAfterMergeWithNull_C2(cond1, cond2, x, y));
-        Asserts.assertEQ(testLoadAfterLoopAlias_Interp(cond1, x, y),                testLoadAfterLoopAlias_C2(cond1, x, y));
+        Asserts.assertEQ(testLoadAfterLoopAlias_Interp(x, y),                       testLoadAfterLoopAlias_C2(x, y));
         Asserts.assertEQ(testCallTwoSide_Interp(cond1, x, y),                       testCallTwoSide_C2(cond1, x, y));
         Asserts.assertEQ(testMergedAccessAfterCallNoWrite_Interp(cond1, x, y),      testMergedAccessAfterCallNoWrite_C2(cond1, x, y));
         Asserts.assertEQ(testCmpMergeWithNull_Second_Interp(cond1, x, y),           testCmpMergeWithNull_Second_C2(cond1, x, y));
-        Asserts.assertEQ(testObjectIdentity_Interp(cond1, 42, y),                   testObjectIdentity_C2(cond1, 42, y));
+        Asserts.assertEQ(testObjectIdentity_Interp(cond1, 42, y),                 testObjectIdentity_C2(cond1, 42, y));
```
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/c2/irTests/scalarReplacement/AllocationMergesTests.java:124]: Wrong spacing.

**Review by vnkozlov — COMMENTED** [review]
Add second (2024) year to Copyright line.

**Review by vnkozlov — APPROVED** [review]
Good.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me too.

**Pull request closed without merging**